sympy/solvers/inequalities.py

```diff
@@ -1007,3 +1010,153 @@ def reduce_inequalities(inequalities, symbols=[]):
 
     # restore original symbols and return
     return rv.xreplace({v: k for k, v in recast.items()})
+
+
+def _pivot(M, i, j):
+    MM = Matrix.zeros(M.rows, M.cols)
+    for ii in range(M.rows):
+        for jj in range(M.cols):
+            if ii == i and jj == j:
+                MM[ii, jj] = 1 / M[i, j]
+            elif ii == i and jj != j:
+                MM[ii, jj] = M[ii, jj] / M[i, j]
+            elif ii != i and jj == j:
+                MM[ii, jj] = -M[ii, jj] / M[i, j]
+            else:
+                MM[ii, jj] = M[ii, jj] - M[ii, j] * M[i, jj] / M[i, j]
+    return MM
+
+
+def _simplex(M, R, S):
+    """
+    Simplex method with randomized pivoting
+    http://web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_LP.pdf
+    """
+    while True:
+        B = M[:-1, -1]
+        A = M[:-1, :-1]
+        C = M[-1, :-1]
+        if all(B[i] >= 0 for i in range(B.rows)):
+            piv_cols = []
+            for j in range(C.cols):
+                if C[j] < 0:
+                    piv_cols.append(j)
+
+            if not piv_cols:
+                return M, R, S
+            random.shuffle(piv_cols)
+            j0 = piv_cols[0]
+
+            piv_rows = []
+            for i in range(A.rows):
+                if A[i, j0] > 0:
+                    ratio = B[i] / A[i, j0]
+                    piv_rows.append((ratio, i))
+
+            if not piv_rows:
+                assert False, 'Unbounded'
```

Comment:
You should redesign these exceptions `assert False, 'Unbounded'`, `assert False, 'Infeasible'` because they are too prototypal.

I just left the design issues to the community, but these should be resolved before this is merged in the end product because it will have backward compatibility issues that makes you cannot change them easier later.

For example, `scipy` wraps the result in an object called `OptimizeResult`
https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.linprog.html
we could do the similar work, but I would favor more minimal design than that